File: recommendation_system/models/movie.py
from datetime import datetime

from models.director import Director


class Movie:
    def __init__(self, title: str, release_date: datetime, director: Director, cast: list, genres: list, rating: float):
        self._title = title
        self._release_date = release_date
        self._director = director
        self._cast = cast
        self._genres = genres
        self._rating = rating

    def get_title(self):
        return self._title

    def set_title(self, title: str):
        self._title = title

    def get_release_date(self):
        return self._release_date

    def set_release_date(self, release_date: datetime):
        self._release_date = release_date

    def get_director(self):
        return self._director

    def set_director(self, director: Director):
        self._director = director

    def get_cast(self):
        return self._cast

    def set_cast(self, cast: list):
        self._cast = cast

    def get_genres(self):
        return self._genres

    def set_genres(self, genres: list):
        self._genres = genres

    def get_rating(self):
        return self._rating

    def set_rating(self, rating: float):
        self._rating = rating

File: movie_handler/src/main/java/com/example/model/Movie.java
package com.example.model;

import com.example.model.enums.Genre;
import com.example.searchService.Search;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;

import java.util.List;
import java.util.UUID;

@Entity
public class Movie {

    @Id
    private UUID id;
    private String title;
    private int year;
    private List<Actor> cast;
    private List<Genre> genre;
    private Director director;
    private double rating;

    public Movie(String title, int year, List<Actor> cast, List<Genre> genre, Director director, double rating) {
        this.title = title;
        this.year = year;
        this.cast = cast;
        this.genre = genre;
        this.director = director;
        this.rating = rating;
    }

    public Movie() {

    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public int getYear() {
        return year;
    }

    public void setYear(int year) {
        this.year = year;
    }

    public List<Actor> getCast() {
        return cast;
    }

    public void setCast(List<Actor> cast) {
        this.cast = cast;
    }

    public List<Genre> getGenre() {
        return genre;
    }

    public void setGenre(List<Genre> genre) {
        this.genre = genre;
    }

    public Director getDirector() {
        return director;
    }

    public void setDirector(Director director) {
        this.director = director;
    }

    public double getRating() {
        return rating;
    }

    public void setRating(double rating) {
        this.rating = rating;
    }

    public void accept(Search searchObject) {

    }
}
